overlord/snapstate/handlers_components.go

```diff
@@ -129,6 +130,82 @@ func (m *SnapManager) doPrepareComponent(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+func (m *SnapManager) doDownloadComponent(t *state.Task, tomb *tomb.Tomb) error {
+	st := t.State()
+	st.Lock()
+	defer st.Unlock()
+
+	compsup, snapsup, err := TaskComponentSetup(t)
+	if err != nil {
+		return err
+	}
+
+	if compsup.CompPath != "" {
+		return fmt.Errorf("internal error: cannot download component %q that specifies a local file path", compsup.ComponentName())
+	}
+
+	if compsup.DownloadInfo == nil {
+		return fmt.Errorf("internal error: cannot download component %q that does not specify download information", compsup.ComponentName())
+	}
+
+	deviceCtx, err := DeviceCtx(st, t, nil)
+	if err != nil {
+		return err
+	}
+
+	user, err := userFromUserID(st, snapsup.UserID)
+	if err != nil {
+		return fmt.Errorf("cannot get user for user ID %d: %w", snapsup.UserID, err)
+	}
+
+	var rate int64
+	if snapsup.IsAutoRefresh {
+		rate = autoRefreshRateLimited(st)
+	}
+
+	cpi := snap.MinimalComponentContainerPlaceInfo(
+		compsup.ComponentName(), compsup.CompSideInfo.Revision,
+		snapsup.InstanceName(),
+	)
+
+	// TODO: to be consistent with snaps, this should be able to point somewhere
+	// else, based on a path that is in the compsup. this would be used for
+	// creating new recovery systems, like it is now for snaps
+	target := cpi.MountFile()
```

Comment:
It should be possible to re-use `snapsup.DownloadBlobDir` for this and implement something similar to `SnapSetup.MountFile()` but for components I guess.

Reply:
Yeah, that would work. I will handle this in a later PR.